src/submission/submission-service.ts

```diff
@@ -352,13 +353,19 @@ export namespace operations {
         if (!newDonorDataMap[donorId]) {
           newDonorDataMap[donorId] = {};
         }
-        newDonorDataMap[donorId][clinicalType] = {
-          ...rc,
-          submitter_donor_id: donorId,
-          index: index,
-        };
+        // by this point we have already validated for uniqueness
+        organizerOperations.addRecord(
```

Comment:
rename `DonorRecordsOrganizerOperations` to `ClinicalSubmissionRecordsOperations`